docs/concepts/ale/ale-driver-architecture.md

```diff
@@ -0,0 +1,147 @@
+            print("mixin_A - method_1")
+            
+        def method_3(self):
+            print("mixin_A - method_3 !")
+            # super().method_3()   # would print "  mixin_B - method_3" if called from class_C
+
+    class mixin_B():
+        def method_2(self):
+            print("mixin_B - method_2")
+        
+        def method_3(self):
+            print("mixin_B - method_3")
+
+        def method_5(self):
+            print("mixin_B - method_5")
+
+    class class_C(mixin_A, mixin_B):
+        def method_4(self):
+            print("class_C - method_4")
+
+        def method_5(self):
+            print("class_C - method_5 !")
+            # super().method_5()   # would print "  mixin_B - method_5"
+            
+    driver = class_C()
+    driver.method_1()
+    driver.method_2()
+    driver.method_3()
+    driver.method_4()
+    driver.method_5()
+    ```
+
+    This example will print:
+    ```
+    mixin_A - method1
+    mixin_B - method2
+    mixin_A - method3 !
+    class_C - method4
+    class_C - method5 !
+    ```
+
+    ## Method Resolution Order
+    In the case of two methods with the same name, 
+    the method from the first leftmost mixin will be used before methods from later mixins, 
+    and the methods in the class overwrite methods from mixins.  
+    `super()` can be used to access methods from parent classes or further-right mixins.
+
+Mixins let ALE drivers mix and match common spacecraft/image data features.
+
+*For example*: There are mixins that correspond to different camera types. 
+Whatever the other mixins and methods in a driver, 
+any methods that correspond to a certain type of camera
+can be included simply by adding a mixin like `Framer`.
+
+More info can be found on ALE Driver mixins in 
+[Creating ALE Drivers - Class Signature](../../how-to-guides/ale-developer-guides/creating-ale-drivers.md/#class-signature).
+
+
+## ale.driver.load()
+
+The `load()` function in ALE is for creating an ISD from a label. 
+
+- `load()` returns a dictionary.
+- `loads()` is a wrapper for `load()` that returns a string.
+- `isd_generate.py` is a user-facing script that uses `load()` and writes a .json file.
+
+`load()` runs through every class containing `_driver` in the name,
+until one works and successfully creates an ISD (or until all fail). 
+It initially checks for an `instrument_id` and fails the driver if that does not succeed. 
+
+Because of the sequential trial-and-error loading method, in some cases
+a different driver may succeed before the desired driver is reached.
+
+
+## Overwriting Base Behavior
+
+The base behavior of a will sometimes need to be overwritten in favor of a more specific behavior for your driver.
+
+!!! note "Use ISIS for Reference!"
+    
+    ISIS is your guide for creating drivers in ALE.
+
+    ALE's ISDs should match closely with with your spacecraft's ISDs from ISIS.
+    A **mismatch** between the ALE and ISIS ISDs (aside from cases of insignificant digits) 
```

Comment:
Similarly to the above, rather than "ALE and ISIS ISDs" I would prefer ALE ISD and ISIS Camera

Reply:
When I wrote this, I was actually thinking of comparing ALE and ISIS labels.  Does this paragraph sound right?

> Labels from cubes generated with an ALE driver should match closely with labels from an ISIS cube.
> A **mismatch** between the ALE and ISIS labels (aside from cases of insignificant digits) 
> is an indication the ALE driver needs to be changed. The mismatched field may give an 
> indication of what method needs implementation or changing.
> 
> A look at related camera classes in ISIS may help show how to implement the ALE Driver.

Reply:
Yes, I think this explains it enough. The only difference you should see is that the ISIS label generated from ALE may have slightly different spice data, but nothing else on the label should be different.

There may be room to talk about USGSCSM cameras vs ISIS cameras else where that would provide a more direct comparison between an ALE ISD and a ISIS Camera